### PR TITLE
Adds Templates to the SDK

### LINKED ISSUE #5: Include Templates in the SDK
Include the tests and code for incorporating templates into the SDK:

https://www.sparkpost.com/docs/templates

### PR DESCRIPTION
**Pull request by iangregory**
Resolves #5 - attempts to remain consistent with the style/tests of the existing code

### CODE REVIEW

File: lib/SparkPost/APIResource.php [hunk 1]
```diff
@@ -177,14 +177,15 @@ private function buildBody($options) {
    * @return array Result set of action performed on resource
    * @throws APIResponseException
    */
-  private function callResource( $action, $resourcePath=null, $options=[] ) {
-    $action = strtoupper($action); // normalize
+  public function callResource( $action, $resourcePath=null, $options=[] ) {
```
Comment (theluk) [inline lib/SparkPost/APIResource.php]: this one may not be public. try `protected` instead, than you can still access it

**Comment by iangregory** [comment]
Thanks @theluk - I've updated the visibility of callResource and also updated the code style to PSR-2 as per the recent php-cs-fixer commit.

**Comment by theluk** [comment]
:+1:

**Comment by richleland** [comment]
Hi @iangregory - users can now use the base resource directly in the 2.x line with any available API. I'm going to go ahead and close this out - but thank you so much for submitting!!

**Pull request closed without merging**